### DISCUSSION TITLE
Run natively on Android? Is it possible?

### DISCUSSION THREAD

**Discussion by pedrib** [NationalSecurityAgency/ghidra#3397] [Ideas]
Is it possible to build Ghidra to run on an Android device?

Weird idea I know - but I have a large 10 inch tablet, and sometimes I prefer to use that instead of my computer when I go to cafes etc.

What would be involved? How hard would that be to achieve?

**Comment by astrelsky** [comment]
> Is it possible to build Ghidra to run on an Android device?
> 
> Weird idea I know - but I have a large 10 inch tablet, and sometimes I prefer to use that instead of my computer when I go to cafes etc.
> 
> What would be involved? How hard would that be to achieve?

Depends on the java version it is running among other OS specific things for the decompiler. There is also the problem of getting it onto the system (unless it is rooted I guess). I highly doubt this would ever be supported and would be a nightmare to use anyway I look at it.

**Comment by ryanmkurtz** [comment]
The easiest way forward is if you follow the instructions in the README and report back what problems you run into.
